- Exception과 그 하위 예외를 말한다.
- 단, RuntimeException과 그 하위 예외는 제외한다.
- 체크 예외를 처리하는 2가지 방법이 존재한다.
	- **예외를 잡아서 처리한다.**
		- `try - catch` 문을 통해 처리할 수 있다.
		- 이때 `catch`는 해당 타입 뿐만 아니라 그 하위 타입도 모두 잡을 수 있다.
		- 해당 클래스를 호출한 상위 클래스로 정상 흐름이 반환된다.
	- **예외를 던진다.**
		- `throws` 문을 통해 던질 수 있다.
		- 체크 예외가 발생했을 경우 상위 클래스까지 예외 흐름이 이어진다.
		- 이때 `throws`는 해당 타입 뿐만 아니라 그 하위 타입도 모두 던질 수 있다.
		- 예외 해결을 다른 클래스에게 위임한다고 볼 수 있다.
- 만약 예외가 `main` 까지 터졌다면 `StackTrace`이 출력되고 애플리케이션이 종료된다.
	- 해당 `StackTrace` 정보를 통해 어디서 문제가 발생했는지 확인해볼 수 있다.
	- 평소에 보는 빨간줄이 그렇게 출력된 것이다.
- 체크 예외의 장단점
	- 개발자가 실수로 예외를 누락하지 않도록 강제하는 좋은 장치이다.
	- 하지만 개발자가 너무 많은 예외를 처리해야 하고, 번거로울 수 있다.
	- 또한 사실상 복구 불가능한 에러의 영역에 가까운 체크 예외인 경우 무의미한 `throws`가 반복될 수 있다.
	- 스트림 API와 람다 표현식에서 체크 예외가 발생하는 코드를 지원하지 않는다.
```java
class Main {  
    public void catchException() {  
	    MyClass myClass = new MyClass();  
        try {  
            myClass.fun("hello");  
        } catch (MyCheckedException e) {
	        System.out.println("예외 발생 : " + e.getMessage())  
        }  
    }
      
    public void throwException() throws MyCheckedException {  
        MyClass myClass = new MyClass();  
        myClass.fun("hello");  
    }  
}  
  
class MyClass {  
    public void fun(String input) throws MyCheckedException {  
        if (input.contains("error")) {  
            throw new MyCheckedException();  
        }  
        System.out.println(input);  
    }  
}  
  
class MyCheckedException extends Exception { }
```